src/box/box.cc

```diff
@@ -3457,7 +3462,18 @@ bootstrap_from_master(struct replica *master)
 	enum applier_state wait_state = replication_anon ?
 					APPLIER_FETCH_SNAPSHOT :
 					APPLIER_INITIAL_JOIN;
-	applier_resume_to_state(applier, wait_state, TIMEOUT_INFINITY);
+	try {
+		applier_resume_to_state(applier, wait_state,
+					TIMEOUT_INFINITY);
+	} catch (...) {
+		return -1;
+	}
+	try {
+		applier_resume_to_state(applier, APPLIER_CONTINUE_SNAPSHOT,
+					TIMEOUT_INFINITY);
```

Comment:
I suggest you refactor this piece a bit.
`wait_state` doesn't matter anymore, because you wait for `APPLIER_CONTINUE_SNAPSHOT` regardless of `wait_state`.

So you may remove the wait for `wait_state` here and only wait for `APPLIER_CONTINUE_SNAPSHOT`